ias-verify/src/tests.rs

```diff
@@ -0,0 +1,145 @@
+const CERT_TOO_SHORT2: &[u8] = b"0\x82\x0c\x8c0";
+
+#[test]
+fn verify_ias_report_should_work() {
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    let report = verify_ias_report(TEST4_CERT);
+    let report = report.unwrap();
+    assert_eq!(report.mr_enclave, TEST4_MRENCLAVE);
+    assert!(report.timestamp >= TEST1_TIMESTAMP.try_into().unwrap());
+    assert_eq!(report.pubkey, TEST4_SIGNER_PUB);
+    //assert_eq!(report.status, SgxStatus::GroupOutOfDate);
+    assert_eq!(report.status, SgxStatus::ConfigurationNeeded);
+}
+
+#[test]
+fn verify_zero_length_cert_returns_err() {
+    // CERT empty, argument 2 and 3 are wrong too!
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    assert!(verify_ias_report(&Vec::new()[..]).is_err())
+}
+
+#[test]
+fn verify_wrong_cert_is_err() {
+    // CERT wrong, argument 2 and 3 are wrong too!
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    assert!(verify_ias_report(CERT_WRONG_PLATFORM_BLOB).is_err())
+}
+
+#[test]
+fn verify_wrong_fake_enclave_quote_is_err() {
+    // quote wrong, argument 2 and 3 are wrong too!
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    assert!(verify_ias_report(CERT_FAKE_QUOTE_STATUS).is_err())
+}
+
+#[test]
+fn verify_wrong_sig_is_err() {
+    // sig wrong, argument 2 and 3 are wrong too!
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    assert!(verify_ias_report(CERT_WRONG_SIG).is_err())
+}
+
+#[test]
+fn verify_short_cert_is_err() {
+    let signer_attn: [u32; 16] = Decode::decode(&mut TEST1_SIGNER_ATTN).unwrap();
+    assert!(verify_ias_report(CERT_TOO_SHORT1).is_err());
+    assert!(verify_ias_report(CERT_TOO_SHORT2).is_err());
+}
+
+#[test]
+fn safe_indexing_works() {
+    let data: [u8; 7] = [0, 1, 2, 3, 4, 5, 6];
+    assert_eq!(safe_indexing(&data, 1, 7), Ok(&data[1..7]));
+    assert_eq!(safe_indexing_one(&data, 3), Ok(3));
+    assert!(safe_indexing(&data, 1, 8).is_err());
+    assert!(safe_indexing(&data, 6, 1).is_err());
+    assert!(safe_indexing(&data, 16, 19).is_err());
+}
+
+#[test]
+fn fix_incorrect_handling_of_iterator() {
+    // In `verify_ias_report` we called `iter.next()` with unwrap three times, which could fail
+    // for certain invalid reports as the one in this test. This test verifies that the issue
+    // has been fixed.
+    //
+    // For context, see: https://github.com/integritee-network/pallet-teerex/issues/35
+
+    let report: [u8; 56] = [
+        224, 224, 224, 224, 224, 224, 224, 224, 235, 2, 0, 1, 5, 40, 0, 8, 255, 6, 8, 42, 134, 72,
+        206, 61, 3, 1, 7, 0, 2, 183, 64, 48, 48, 0, 1, 10, 23, 3, 6, 9, 96, 134, 72, 1, 134, 248,
+        66, 1, 13, 0, 0, 0, 13, 1, 14, 177,
+    ];
+
+    assert_err!(verify_ias_report(&report), "Invalid netscape payload");
+}
```

Comment:
Test the fix.

Reply:
nice 🚀

Reply:
I took this one from #35.

If I needed to generate one, I would print  a report by the enclave and copy-paste it here.